### PR TITLE
MRTCore take ms-resource URI

### PR DESCRIPTION
**Pull request by huichen123**
WinUI3 needs to pass in full URI in some cases. Make MRTCore handle that. When full URI is provided, the parent resource map is ignored.

### CODE REVIEW

**Review by axelandrejs — APPROVED** [review]

File: dev/MRTCore/mrt/Core/src/MRM.cpp [hunk 1]
```diff
@@ -28,12 +28,55 @@ typedef struct _MrmObjects
...
+    return ((wcslen(resourceId) > static_cast<size_t>(ResourceUriPrefixLength)) && (CompareStringOrdinal(ResourceUriPrefix, ResourceUriPrefixLength, resourceId, ResourceUriPrefixLength, TRUE) == CSTR_EQUAL));
+}
+
+HRESULT GetRootAndRelativeIdFromUri(
+    _In_ PCWSTR resourceUri,
+    _Out_writes_z_(rootSize) PWSTR root,
+    unsigned int rootSize,
+    _Out_ unsigned int* indexOfRelatvieId)
+{
+    // We should have checked resourceUri length is longer than ResourceUriPrefixLength before calling this
```
Comment (robertzhou-wpn) [inline dev/MRTCore/mrt/Core/src/MRM.cpp:48]: Wondering where we do this check?

Comment (huichen123) [reply dev/MRTCore/mrt/Core/src/MRM.cpp:48]: The function is always called after IsResourceUri. The check is done in IsResourceUri.

**Pull request merged**